### PR TITLE
throw error while assigning to a subroutine

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fix https://github.com/lfortran/lfortran/issues/1878

### CODE REVIEW

File: tests/errors/subroutine1.f90 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+program subroutine1
+integer :: i
+call bpe()
+
+contains
+
+subroutine bpe()
+bpe = 1
+end subroutine
+end program
```
Comment (Shaikh-Ubaid) [inline tests/errors/subroutine1.f90:10]: - Please if possible use indentation(s). It makes the code/test look good. 
- Please remove the variable `i` declaration, if it is unused. I think having a minimum code that tests a given feature is good.

**Review by certik — APPROVED** [review]
Excellent, great fix!

File: tests/errors/subroutine1.f90 (hunk 1 above)
Comment (certik) [reply tests/errors/subroutine1.f90:10]: My apologies, I missed your comment, you posted it while I was reviewing it also. @HarshitaKalani, you can fix that in a subsequent PR.

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 2]
```diff
@@ -2097,6 +2097,9 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                                 }));
                 }
             }
+            if(sym->type == ASR::symbolType::Function){
```
Comment (certik) [inline src/lfortran/semantics/ast_body_visitor.cpp:2100]: I recommend using:
```
if (ASR::is_a<ASR::Function_t>(*sym))
```
Which does some additional checks and it's independent how the type itself is implemented underneath.

I know you copied the style from above and it should be fixed above as well.

File: tests/errors/subroutine1.f90 (hunk 1 above)
Comment (HarshitaKalani) [reply tests/errors/subroutine1.f90:10]: Sure, I'll do that. Thanks.

**Pull request merged**